app/models/watchlist_instance.rb

```diff
@@ -22,6 +22,38 @@ def self.get_num_new_instance_for_course(course_name)
     return WatchlistInstance.where(course_id:course_id,status: :new).count()
   end 
 
+  def self.refresh_instances_for_course(course_name)
+    begin
+      course = Course.find_by(name: course_name)
+      current_conditions = RiskCondition.get_current_for_course(course_name)
+      current_instances = WatchlistInstance.where(course_id: course.id)
+    rescue NoMethodError
+      raise "Course #{course_name} cannot be found"
+    end
+
+    # check whether the watchlist instances are up-to-date
+    current_instances_condition_ids = (current_instances.map { |inst| inst.risk_condition_id }).uniq.to_set
+    current_condition_ids = (current_conditions.map { |c| c.id }).uniq.to_set # should be unique by design but add the suffix to ensure
```

Comment:
This condition check for to reduce calculation actually causes a problem. Because if the conditions are kept the same, but the assignment deadlines etc are changed, this will actually prevent the instructor from updating the watchlist.

Reply:
Removing this check solved my issue